### DISCUSSION TITLE
Is there a `tsc` equivalent in deno ?

### DISCUSSION THREAD

**Discussion by yw662** [denoland/deno#8842] [Q&A] [answered]
Like, I mean, if there is no node and tsc installed, can I use deno to compile some typescripts into javascript, just in the same way `tsc` compile files ?

**Comment by kitsonk** [comment, marked as answer]
The runtime compiler APIs provided this which is covered in the manual here: https://deno.land/manual@v1.6.1/runtime/compiler_apis.

They are _unstable_ and will be rewritten, like to land in Deno 1.7.0 via #8799.

**Reply by yw662** [reply]
But is there a somehow standardized way to write the compiled files to disk keeping their original relative path if not using `.bundle` ?

**Comment by shafkathullah** [comment]
@yw662  I think you are asking about type checking. compiling/transpiling TS to JS is easier than type checking(tools like swc already does this), as far as I know only tsc is able to type check TS files. there was an attempt by swc author called swt or something which i think he dropped.
